components/JobsSidebar.vue

```diff
@@ -108,6 +115,13 @@ export default {
       } else {
         $state.complete()
       }
+    },
+    intersected() {
+      console.log('Getting more data...')
+
+      if (this.show < this.jobs.length) {
```

Comment:
Here you're putting the burden of deciding whether to continue or stop on the parent component.  I quite like the original's approach of loaded() vs finished() as a way of moving that logic out of the parent.  

It could be even better, though:
* A clear way to wake the component up again later when things change.
* Some generic logic for handling the case where you've fetched some stuff and run out.  That needs some thought but we do it all over so it would be worth centralising.

Some of this might do in a higher level component than this one, ie this is a simple observer-wrapper and then there is infinite scroll logic which uses it.

Reply:
I've just copied this logic from what it did before.  This was the previous method:

```
if (this.show < this.jobs.length) {
        this.show += 3
        $state.loaded()
      } else {
        $state.complete()
      }
```

There is definitely some improvements that could be made in this area.  I imagine that even when all the data is loaded that the observer will still run at the bottom of the list so we could detach it maybe.  I'm guessing that is what you mean by your second bullet point.

Reply:
No, I meant something else but it wasn't clear.  

Look at modMessagesPAge around line 139.  We get currentCount, then we fetch, then we check the currentCount against the (new) computed property from the store to decide whether we should call complete or loaded.   We have a bunch of logic like that which is a bit poor.

But on reflection that's part of having a better interface to the scroll, and perhaps a better interface from the server.  So ignore it for now.